src/arithmetic/arithmetic_expression_construct.c

```diff
@@ -562,6 +654,8 @@ static AR_ExpNode *_AR_EXP_FromASTNode(const cypher_astnode_t *expr) {
 		return _AR_ExpFromSliceExpression(expr);
 	} else if(t == CYPHER_AST_NAMED_PATH) {
 		return _AR_ExpFromNamedPath(expr);
+	} else if(t == CYPHER_AST_SHORTEST_PATH) {
```

Comment:
I think we can collapse this entire if else block into a simple array access,
e.g. `ARRAY[CYPHER_AST_SHORTEST_PATH](expr);` 

see deps/libcypher-parser/lib/src/ast.c : 
```
const uint8_t CYPHER_AST_SHORTEST_PATH = VT_OFFSET(shortest_path);
```

Reply:
This seems like a good idea to me, the array would only be of size 256. The only nuisance is that the offset values aren't known at compile-time, so we'd need to populate the array during module initialization.